2_process/src/create_obs_lake_task_plan.R

```diff
@@ -0,0 +1,128 @@
+      sprintf("filtered_obs_data_%s", task_name)
+    },
+    command = function(task_name, ...){
+      task_filepath <- dplyr::filter(tasks, site_id == task_name) %>% pull(task_filepath)
+      sprintf("filter_observed_data('%s')", task_filepath)
+    } 
+  )
+  
+  reformat_obs_data <- scipiper::create_task_step(
+    step_name = 'reformat_obs_data',
+    target_name = function(task_name, ...){
+      sprintf("formatted_obs_data_%s", task_name)
+    },
+    command = function(steps, ...){
+      sprintf("reformat_observed_data(`%s`)", steps[["apply_data_criteria"]]$target_name)
+    } 
+  )
+  
+  join_temp_kd_i0 <- scipiper::create_task_step(
+    step_name = 'join_temp_kd_i0',
+    target_name = function(task_name, step_name, ...){
+      sprintf("2_process/tmp/munged_obs_data_%s.feather", task_name)
+    },
+    command = function(task_name, steps, ...){
+      irr_filepath <- dplyr::filter(tasks, site_id == task_name) %>% pull(irr_filepath)
+      clarity_filepath <- dplyr::filter(tasks, site_id == task_name) %>% pull(clarity_filepath)
+      psprintf("join_observed_data(",
+               "target_name = target_name,",
+               "obs_data = `%s`," = steps[["reformat_obs_data"]]$target_name, 
+               "irr_data_fn = '%s'," = irr_filepath, 
+               "k0_data_fn = '%s')" = clarity_filepath)
+    } 
+  )
+  
+  calculate_obs_toha <- scipiper::create_task_step(
+    step_name = 'calculate_obs_toha',
+    target_name = function(task_name, ...){
+      sprintf("2_process/tmp/obs_toha_%s.csv", task_name)
+    },
+    command = function(task_name, steps, ...){
+      psprintf("calculate_toha_per_lake(", 
+               "target_name = target_name,",
+               "site_data_fn = '%s'," = steps[["join_temp_kd_i0"]]$target_name,
+               "morphometry = `%s_morphometry`)" = task_name
+      )
+    } 
+  )
+  
+  # ---- combine into a task plan ---- #
+  
+  task_plan <- create_task_plan(
+    task_names = tasks$site_id,
+    task_steps = list(
+      split_morphometry, apply_data_criteria, reformat_obs_data,
+      join_temp_kd_i0, calculate_obs_toha),
+    final_steps = 'calculate_obs_toha',
+    add_complete = FALSE)
+  
+  # ---- combine into a task makefile ---- #
+  
+  create_task_makefile(
+    task_plan = task_plan,
+    makefile = '2_obs_lake_tasks.yml',
+    include = 'remake.yml',
+    sources = c(...),
+    packages = c("purrr", "dplyr", "mda.lakes", "feather", "rLakeAnalyzer", "readr"),
+    final_targets = c(target_name),
+    finalize_funs = c('combine_obs_toha'),
+    as_promises = TRUE,
+    tickquote_combinee_objects = TRUE)
+  
+  # ---- build the task makefile ---- #
+  
+  loop_tasks(
+    task_plan = task_plan,
+    task_makefile = '2_obs_lake_tasks.yml',
+    num_tries = 1)
+  
+}
+
+combine_obs_toha <- function(target_name, ...) {
+  # Read feather files and combine into single CSV
+  purrr::map(list(...), read_csv, col_types = cols()) %>% 
+    purrr::reduce(bind_rows) %>% 
```

Comment:
See comment below on `reduce` vs a direct call to `c` - because `bind_rows` accepts an unlimited number of arguments whereas `reduce` is optimized for functions that only accept two arguments, I think calling `bind_rows` directly would be more efficient.

Reply:
Huh, this approach is one I've seen used many times. I hadn't even considered not using `reduce` with `bind_rows`. @jread-usgs I think I got this pattern from you so want to make you aware, too

Reply:
Yeah, I didn't know how best to use it, so good to know

Reply:
Quick test below. Unlike `c` ([see here](https://github.com/USGS-R/lake-temperature-out/pull/27#discussion_r451785073)), `bind_rows` does operate in this way.

```r
library(dplyr)
mylistdf <- list(
  one = tibble(x = 1:3, y=4:6),
  two = tibble(x = 4:6, y=7:9),
  three = tibble(x = 7:9, y=10:12)
)

purrr::reduce(mylistdf, bind_rows)
# A tibble: 9 x 2
      x     y
  <int> <int>
1     1     4
2     2     5
3     3     6
4     4     7
5     5     8
6     6     9
7     7    10
8     8    11
9     9    12

bind_rows(mylistdf)
# A tibble: 9 x 2
      x     y
  <int> <int>
1     1     4
2     2     5
3     3     6
4     4     7
5     5     8
6     6     9
7     7    10
8     8    11
9     9    12

mylistdf %>% bind_rows()
# A tibble: 9 x 2
      x     y
  <int> <int>
1     1     4
2     2     5
3     3     6
4     4     7
5     5     8
6     6     9
7     7    10
8     8    11
9     9    12
```